first-react-app/src/components/homeComponents/CardsShowcase.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import Card from "./Card";
+import ModalCard from "./ModalCard";
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch, RootState } from "../../data/reduxStore";
+import { changeModal } from "../../data/artistModalCardSlice";
+
+function Cards() {
+  const artistsStore = useSelector((state: RootState) => state.artistsInStore);
+  const modalStore = useSelector((state: RootState) => state.artistsInfoStore.artistsInfo);
+  const isModalActiveStore = useSelector(
+    (state: RootState) => state.artistsInfoStore.isModalActive
+  );
+  const dispatch = useDispatch<AppDispatch>();
+
+  function closeModalWindow() {
+    dispatch(changeModal(false));
+  }
+
+  return (
+    <>
+      <div className="modal-wrapper">
+        {modalStore && isModalActiveStore ? (
+          <ModalCard
+            key={Math.random()}
+            handleModalClose={closeModalWindow}
+            name={modalStore!.name}
+            similar={modalStore!.similar}
+            tags={modalStore!.tags}
+            bio={{
+              published: modalStore!.published,
+              summary: modalStore!.summary,
+            }}
+          ></ModalCard>
+        ) : null}
+      </div>
+      <h1>{artistsStore.status}</h1>
+      <div className="cards-wrapper">
+        {artistsStore.artists?.map((el) => {
+          return <Card songArtist={el.name} songImage={el.image} key={Math.random()}></Card>;
```

Comment:
do not generate keys on the fly, e.g. with `key={Math.random()}`. This will cause keys to never match up between renders, leading to all your components and DOM being recreated every time. Not only is this slow, but it will also lose any user input inside the list items. Instead, use a stable ID based on the data.
https://react.dev/learn/rendering-lists#why-does-react-need-keys
https://legacy.reactjs.org/docs/lists-and-keys.html#keys
```suggestion
          return <Card songArtist={el.name} songImage={el.image} key={el.id}></Card>;
```